### PR TITLE
Minimal 4.13 support

### PR DESCRIPTION
**Pull request by Octachron**
This PR makes odoc build on OCaml 4.13 but does not handle the new `with module type` feature.
This PR mostly adapts the loaders code to the few ASTs change in 4.13 .
There are also few fixed related to the change of typing for `|>`.

### CODE REVIEW

File: src/xref2/env.ml [hunk 1]
```diff
@@ -272,35 +272,35 @@ let add_type identifier t env =
   |> add_cdocs identifier t.doc
   |> List.fold_right (add_cdocs identifier) docs
 
-let add_module_type identifier t env =
+let add_module_type identifier (t:Component.ModuleType.t) env =
```
Comment (Julow) [inline src/xref2/env.ml:275]: Why are these needed ? There's an mli

File: src/html/generator.ml [hunk 2]
```diff
@@ -292,9 +292,9 @@ and items ~resolve l : item Html.elt list =
             | _ -> Stop_and_keep)
         in
         let content = flow_to_item @@ block ~resolve text in
-        content |> (continue_with [@tailcall]) rest
+        (continue_with [@tailcall]) rest content
     | Heading h :: rest ->
-        [ heading ~resolve h ] |> (continue_with [@tailcall]) rest
+         (continue_with [@tailcall]) rest [ heading ~resolve h ]
```
Comment (Julow) [inline src/html/generator.ml:297]: Why is this needed ?

File: src/loader/cmti.ml [hunk 3]
```diff
@@ -470,6 +470,10 @@ let rec read_with_constraint env parent (_, frag, constr) =
         let frag = Env.Fragment.read_module frag.Location.txt in
         let p = Env.Path.read_module env p in
           ModuleSubst(frag, p)
+#if OCAML_VERSION >= (4,13,0)
+   | Twith_modtype _ -> failwith "with module type not yet implemented"
```
Comment (Julow) [inline src/loader/cmti.ml:474]: We need to do this before 2.0: https://github.com/ocaml/odoc/issues/688

File: src/loader/cmt.ml [hunk 4]
```diff
@@ -48,7 +48,11 @@ let rec read_pattern env parent doc pat =
     | Tpat_constant _ -> []
     | Tpat_tuple pats ->
         List.concat (List.map (read_pattern env parent doc) pats)
+#if OCAML_VERSION < (4, 13, 0)
```
Comment (Julow) [inline src/loader/cmt.ml:51]: :+1:  for the tuple

**Comment by jonludlam** [comment]
Looks like just the format test failed.

**Review by Julow — APPROVED** [review]

File: src/html/generator.ml (hunk 2 above)
Comment (Julow) [reply src/html/generator.ml:297]: You answered that already.

File: src/xref2/env.ml (hunk 1 above)
Comment (Octachron) [reply src/xref2/env.ml]: The existence of a mli is irrelevant? The `doc` field is not in scope, the previous code was relying on the specific order of type propagation to resolve the field, this was brittle. Adding a type annotation on `t` fixes the issue.

Comment (Julow) [reply src/xref2/env.ml]: Indeed.
